### DISCUSSION TITLE
SteamCMD is not online?

### DISCUSSION THREAD

**Discussion by zephiowl** [wolveix/satisfactory-server#153] [Q&A] [answered]
Hey so i'm attempting to run the docker instance for the first time on my Truenas Scale homelab. 
With this run command:

`docker run -d --name=satisfactory-server -h satisfactory-server -e MAXPLAYERS=4 -e PGID=1000 -e PUID=1000 -e STEAMBETA=false -v /mnt/raid-ssd/games/Satisfactory/config:/config -p 7777:7777/udp -p 15000:15000/udp -p 15777:15777/udp wolveix/satisfactory-server:latest`
And i keep getting an error saying steamcmd needs to be online to update. 
Logs here:

```
Checking available memory...12GB detected
Setting max objects number to 2162688
Setting timeout number to 300
Setting autosave number to 5
Setting crash reporting to True
Setting max players to 4
Setting autosave interval to 300s
Setting disable seasonal events to 0
Setting network quality number to 3
Setting auto pause to True
Setting autosave on disconnect to True
Checking available space...99GB detected
Downloading the latest version of the game...
Redirecting stderr to '/home/steam/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
[  0%] Downloading update...
[  0%] Checking for available updates...
[----] !!! Fatal Error: Steamcmd needs to be online to update. Please confirm your network connection and try again.
src/tier0/threadtools.cpp (3628) : Assertion Failed: Illegal termination of worker thread 'Thread(0x0x56cb7fd0/0x0xf771db'
```

Any suggested fixes?

**Comment by wolveix** [comment]
Hey! Probably sounds like a silly question, but does the container definitely have internet access? Are you running on a Linux host?

**Reply by zephiowl** [reply]
I believe so; I'm currently running the instance off trunas scale. Which uses debian as its distro and I'm on Kernel version 5.10.142. 
![image](https://user-images.githubusercontent.com/66909997/200152139-b1da922a-ad97-4424-9e28-b308f2318ac7.png)

I'm able to run other docker instances perfectly fine such as a dockerized [Minecraft server](https://hub.docker.com/r/itzg/minecraft-server) a [Project Zomboid server](https://hub.docker.com/r/afey/zomboid) and a customizable [Discord bot](https://hub.docker.com/r/phasecorex/red-discordbot) all with little to no issues. I'd be more then happy to send some more info over if you'd like but I dont believe that connecting to the internet is the issue. As much as the logs want to say otherwise. Or maybe its something spesific to this container? I'm not sure if it would be a steamcmd issue as I've been able to get other steam games like [The Forrest](https://hub.docker.com/r/jammsen/the-forest-dedicated-server) or [Project Zomboid server](https://hub.docker.com/r/afey/zomboid) to run perfectly fine in the past.

**Reply by wolveix** [reply]
I really appreciate your thorough reply! That's really interesting. If you use Discord, would you mind adding me there? It's generally easier to have a back and forth discussion there, and then report our findings here for anyone else who may experience this.

If not, no problem! I can convert this into an issue and we can continue there :)

**Reply by zephiowl** [reply]
No problem at all; matrix and Discord are some of my main social platforms. I've gone ahead and sent Wolveix#2211 A friend request (found off your readme.md) It should be coming from a Xanthus#3862

**Comment by zephiowl** [comment, marked as answer]
After about an hour or two of headache me and @wolveix where able to figure out its an issue with IX-based machines/operating systems. I'm not 100% sure on it but it seemed to be that the issue was manually mapping ports with docker. Removing all -p flags and replacing it with --net=host seems to have fixed the issue.
